contracts/src/StakeTable.sol

```diff
@@ -212,8 +286,19 @@ contract StakeTable is AbstractStakeTable {
     /// @param blsVK The BLS verification key to withdraw
     /// @return The total amount withdrawn, equal to `Node.balance` associated with `blsVK`
     function withdrawFunds(BN254.G2Point memory blsVK) external override returns (uint64) {
-        bytes32 hash = _hashBlsKey(blsVK);
-        nodes[hash].balance = 0;
-        return 0;
+        bytes32 key = _hashBlsKey(blsVK);
+        Node memory node = nodes[key];
+
+        if (currentEpoch() < node.exitEpoch + exitEscrowPeriod(node)) {
+            revert PrematureWithdrawal();
+        }
+        uint64 balance = node.balance;
+        SafeTransferLib.safeTransfer(ERC20(tokenAddress), node.account, balance);
+
+        Node memory nullNode =
+            Node(address(0), StakeType.Native, 0, 0, 0, EdOnBN254.EdOnBN254Point(0, 1));
+        nodes[key] = nullNode;
```

Comment:
can simply use `delete nodes[key]` syntax.

Reply:
Right, done in 978171a4744d748d21993b63809a51018b3642f8.